### PR TITLE
add multisite functionality

### PR DESCRIPTION
**Pull request by codiebeulaine**

### CODE REVIEW

**Comment by codiebeulaine** [comment]
@miltontony @saeedmb8 ready for review

File: molo/core/middleware.py [hunk 1]
```diff
@@ -135,3 +139,17 @@ def process_response(self, request, response):
                 site_settings.global_ga_tracking_code, request, response)
 
         return response
+
+
+class MultiSiteRedirectToHomepage(object):
+
+    def process_request(self, request):
+        if request.path.startswith('/admin/pages/'):
```
Comment (miltontony) [inline molo/core/middleware.py]: You said you had an issue with `/import/`, how did you fix it?

File: molo/core/tasks.py [hunk 2]
```diff
@@ -25,20 +24,21 @@ def rotate_content(day=None):
...
-        rotate_latest(main_lang, index, main, site_settings, day)
-        rotate_featured_in_homepage(main_lang, day)
+
+    for main in Main.objects.all():
+        site = main.sites_rooted_here.all().first()
+        main_lang = Languages.for_site(site).languages.filter(
+            is_main_language=True).first()
+        index = SectionIndexPage.objects.live().child_of(main).first()
+        settings2 = SettingsProxy(site)
+        site_settings = settings2['core']['SiteSettings']
```
Comment (miltontony) [inline molo/core/tasks.py]: shouldn't this be `SiteSettings.for_site(site)`?

File: molo/core/tests/base.py [hunk 3]
```diff
@@ -79,13 +73,67 @@ def mk_main(self):
...
+
+    def mk_main2(self, title='main2', slug='main2', path=00010002):
+        # Create page content type
+        page_content_type, created = ContentType.objects.get_or_create(
+            model='page',
+            app_label='wagtailcore'
+        )
+
+        # Create root page
+        self.root, _ = Page.objects.get_or_create(
```
Comment (miltontony) [inline molo/core/tests/base.py]: I see a lot of this cruff - can we just put this in base?
```python
self.root = self.mk_root_page()
```

File: molo/core/middleware.py (hunk 1 above)
Comment (codiebeulaine) [inline molo/core/middleware.py]: The issue wasn't this, I didn't actually fix it, I as fixing other tests and those passed as a result. Not sure what made it pass

File: molo/core/tests/base.py (hunk 3 above)
Comment (codiebeulaine) [inline molo/core/tests/base.py]: yeah

File: molo/core/tasks.py (hunk 2 above)
Comment (codiebeulaine) [inline molo/core/tasks.py]: both works, will change to the for_site

**Comment by miltontony** [comment]
👍 🍰 🎈 🌟 🎆 wtih

**Pull request merged**